### PR TITLE
sssd remove use of python-pam

### PR DESCRIPTION
**Pull request by tonyyuan1**
sssd test used python-pam, however it's unmantained and it was removed from package hub repo
- Related ticket: https://progress.opensuse.org/issues/89479
- Needles: no
- Verification run: 
- sle15sp3: http://10.67.17.201/tests/1401
- sle15sp2: http://10.67.17.201/tests/1386
- sle15sp1: http://10.67.17.201/tests/1383
- sle12sp5: http://10.67.17.201/tests/1400
- tumbleweed: http://10.67.17.201/tests/1399

### CODE REVIEW

File: tests/sysauth/sssd.pm [hunk 1]
```diff
@@ -36,20 +36,13 @@ use version_utils qw(is_sle is_opensuse);
...
     $self->select_serial_terminal;
-
     # Install test subjects and test scripts
     my @test_subjects = qw(
       sssd sssd-krb5 sssd-krb5-common sssd-ldap sssd-tools
       openldap2 openldap2-client
       krb5 krb5-client krb5-server krb5-plugin-kdb-ldap
     );
 
-    # for sle 12 we still use and support python2
```
Comment (dzedro) [inline tests/sysauth/sssd.pm:47]: Update also test description

Comment (tonyyuan1) [reply tests/sysauth/sssd.pm:47]: Good catch! Updated.

**Review by foursixnine — CHANGES_REQUESTED** [review]
You are removing all the testing done via PAM.

`In the end, this might be a complete rewrite of the sssd tests, rendering tests/sysauth/sssd.pm useless. A bit of background, those tests are a bit extensive but they were written by a developer that is long gone from the company, investing time in fixing or looking for an alternative to python-pam and rewrite the test suite is more expensive than switching to directly testing the service against the system (by means of directly using pam/passwd/getent)`

You can adapt these tests to use directly openldap + sssd + pam via `getent` for instance to test the autentication.

But I'd propose a full module rewrite with a much simpler approach (like the pr mentioned in the ticket)

**Comment by tonyyuan1** [comment]
> You are removing all the testing done via PAM.
> 
> `In the end, this might be a complete rewrite of the sssd tests, rendering tests/sysauth/sssd.pm useless. A bit of background, those tests are a bit extensive but they were written by a developer that is long gone from the company, investing time in fixing or looking for an alternative to python-pam and rewrite the test suite is more expensive than switching to directly testing the service against the system (by means of directly using pam/passwd/getent)`
> 
> You can adapt these tests to use directly openldap + sssd + pam via `getent` for instance to test the autentication.
> 
> But I'd propose a full module rewrite with a much simpler approach (like the pr mentioned in the ticket)

This test uses some "su" commands which use PAM but they run su as root to verify password.  It seems incorrect.  root can su any user without password. I did not try them manually.  
I like your proposal to rewrite it.  I can rewrite it after hackweek. This PR is no use.  I will close it if you have no objection.

**Comment by foursixnine** [comment]
@tonyyuan1 sure thing :) Thanks!

**Pull request closed without merging**